src/app/clusters/ota-requestor/DefaultOTARequestorDriver.cpp

```diff
@@ -158,8 +199,10 @@ void DefaultOTARequestorDriver::UpdateAvailable(const UpdateDescription & update
         delay, [](System::Layer *, void * context) { ToDriver(context)->mRequestor->DownloadUpdate(); }, this);
 }
 
-void DefaultOTARequestorDriver::UpdateNotFound(UpdateNotFoundReason reason, System::Clock::Seconds32 delay)
+CHIP_ERROR DefaultOTARequestorDriver::UpdateNotFound(UpdateNotFoundReason reason, System::Clock::Seconds32 delay)
```

Comment:
I think you could utilize the newly introduced `ScheduleRetry` for the Busy/NotAvailable case? Something like:

```
    switch (reason)
    {
    case UpdateNotFoundReason::kUpToDate:
        willTryAnotherQuery = false;
        break;
    case UpdateNotFoundReason::kBusy:
        status = ScheduleRetry(true);
        if (status == CHIP_NO_ERROR)
        {
            break;
        }
        else if (status != CHIP_ERROR_MAX_RETRY_EXCEEDED)
        {
            // Max retry has not been exceeded but there are other issues with scheduling the retry - probably do nothing for now?
            break;
        }
        ChipLogProgress(SoftwareUpdate, "Max Busy Provider retries reached. Attempting to get next Provider.");
        __attribute__((fallthrough)); // fallthrough
    case UpdateNotFoundReason::kNotAvailable: {
        // IMPLEMENTATION CHOICE:
        // This implementation schedules a query only if a different provider is available
        // Note that the "listExhausted" being set to TRUE, implies that the entire list of
        // defaultOTAProviders has been traversed. On bootup, the last provider is reset
        // which ensures that every QueryImage call will ensure that the list is traversed from
        // start to end, until an OTA is successfully completed.
        status = ScheduleRetry(false);
        if (status == CHIP_NO_ERROR)
        {
            break;
        }
        if (status == SOME_ERROR_INDICATING_LIST_EXHAUSTED)
        {
            break;
        }
        break;
```

Reply:
Made the change to use ScheduleQueryRetry().